src/coreclr/vm/interpexec.cpp

```diff
@@ -1200,11 +1285,83 @@ void InterpExecMethod(InterpreterFrame *pInterpreterFrame, InterpMethodContextFr
                     UNREACHABLE();
                     break;
                 }
+
+                case INTOP_ENTER_FUNCLET:
```

Comment:
I don't think there is any reason for `INTOP_ENTER_FUNCLET` to exist and to pass throwable as an argument to interp exec method. When invoking a clause, EH could write directly to the stack offset where the exception object is expected.

Reply:
This opcode helped me with a few things:

1. This allow all funclet blocks to have an empty IL evaluation stack to start with - that also made sure all the funclet entry blocks are generated on the first pass without retry. 
2. This allow me to decouple the IR generator from the interpreter by not having to describe where the exception is stored.
3. This allow me to use a local slot to store the exception object instead of a global one, that space is likely reused for the non-exceptional path.

Not sure what is the advantage of doing otherwise.

Reply:
In general we should aim to have as few opcodes generated as possible for the sake of performance. In this particular example, given this is only used in the exceptional path for filter and catch funclets, I'd say it is ok to leave as is, given 2 and 3 represent good enough motivation.